test/suite/test_compact09.py

```diff
@@ -0,0 +1,169 @@
+        return files_compacted
+
+    def get_pages_rewritten(self, uri):
+        stat_cursor = self.session.open_cursor('statistics:' + uri, None, None)
+        pages_rewritten = stat_cursor[stat.dsrc.btree_compact_pages_rewritten][2]
+        stat_cursor.close()
+        return pages_rewritten
+
+    def populate(self, uri, num_keys, value_size):
+        c = self.session.open_cursor(uri, None)
+        for k in range(num_keys):
+            c[k] = ('%07d' % k) + '_' + 'abcd' * ((value_size // 4) - 2)
+        c.close()
+
+    def turn_off_bg_compact(self):
+        self.session.compact(None, 'background=false')
+        compact_running = self.get_bg_compaction_running()
+        while compact_running:
+            time.sleep(1)
+            compact_running = self.get_bg_compaction_running()
+        self.assertEqual(compact_running, 0)
+
+    def turn_on_bg_compact(self, config):
+        self.session.compact(None, config)
+        compact_running = self.get_bg_compaction_running()
+        while not compact_running:
+            time.sleep(1)
+            compact_running = self.get_bg_compaction_running()
+        self.assertEqual(compact_running, 1)
+
+    # Test the exclude list functionality of the background compaction server. 
+    def test_compact09(self):
+        # FIXME-WT-11399
+        if self.runningHook('tiered'):
+            self.skipTest("this test does not yet work with tiered storage")
+
+        # Create and populate tables.
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            self.session.create(uri, self.create_params)
+            self.populate(uri, self.table_numkv, self.value_size)
+        
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete the first 90%.
+        for i in range(self.n_tables):
+            uri = self.uri_prefix + f'_{i}'
+            self.delete_range(uri, 90 * self.table_numkv // 100)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Enable background compaction and exclude the two tables.
+        exclude_list = f'["{self.uri_prefix}_0.wt", "{self.uri_prefix}_1.wt"]'
+        config = f'background=true,free_space_target=1MB,exclude={exclude_list}'
+        self.turn_on_bg_compact(config)
+        
+        # Background compaction should exclude all files.
+        while self.get_bg_compaction_files_excluded() < self.n_tables:
+            time.sleep(1)
+        assert self.get_files_compacted() == 0
+
+        # Stop the background compaction server.
+        self.turn_off_bg_compact()
+
+        # Enable background compaction and exclude only one table.
+        exclude_list = f'["{self.uri_prefix}_0.wt"]'
+        config = f'background=true,free_space_target=1MB,exclude={exclude_list}'
+        self.turn_on_bg_compact(config)
+
+        # Background compaction should exclude only one file now.
+        while self.get_bg_compaction_files_excluded() < self.n_tables + 1:
```

Comment:
It's confusing that we have 2 tables but are now checking that excluded should be less than 3? You should add a comment explaining this check, or reset the connection/stats and check that `self.get_bg_compaction_files_excluded()` < 1

Reply:
Sure! I added a comment in 6bcf14e05060e1471fa356ef6b3717d21d93d1df.